### PR TITLE
Ticket2624 file tracking

### PR DESCRIPTION
**Pull request by ThomasLohnert**
### Description of work

Simplified the system responsible for keeping user scripts and configurations under version control:
VC is now managed by a thread running in the blockserver, which simply adds, commits and pushes all files in the repositories every 5 minutes. This change greatly reduces the complexity of the system, removes the need for any filewatchers, and with it previously problematic interactions between the filewatcher and the version control threads.

### To test

https://github.com/ISISComputingGroup/IBEX/issues/2624

### Acceptance criteria

- [x] Any changes on the file system are committed to git in regular intervals (currently 5 minutes - you can tweak this interval in the `PUSH_BASE_INTERVAL` variable in `git_version_control.py` if it makes testing easier
- [x] The commit messages provide an appropriate summary of changes
- [x] Graceful handling of problematic file system / git interactions: Trying to save a file while it is being added; trying to add a file while it is being saved. You can use the following powershell script to manually lock/unlock a file for testing:
- [x] Interactions with configurations from the GUI are unchanged

```
#Specify the file name
$fileName = "[file path]"

#Open the file in read only mode, without sharing (I.e., locked as requested)
$file = [System.io.File]::Open($fileName, 'Open', 'Read', 'None')

#Wait in the above (file locked) state until the user presses a key
Write-Host "Press any key to continue ..."
$null = $host.UI.RawUI.ReadKey("NoEcho,IncludeKeyDown")

#Close the file (This releases the current handle and unlocks the file)
$file.Close()
```
---

#### Code Review

- [x] Is the code of an acceptable quality?
- [x] Has the author taken into account the multi-threaded nature of the code?
- [x] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?
- [ ] Has the [manual system tests spreadsheet](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Manual-system-tests) been updated?

### Functional Tests

- [x] Do changes function as described? Add comments below that describe the tests performed.

### Final steps

- [ ] Reviewer has updated the submodule in the main EPICS repo? See **Reviewing work for the subModules of EPICS** in the [Git workflow](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Git-workflow) page for details.
- [x] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: ConfigVersionControl/git_version_control.py [hunk 1]
```diff
@@ -15,19 +15,18 @@
 # http://opensource.org/licenses/eclipse-1.0.php
 
 import os
```
Comment (Tom-Willemsen) [inline ConfigVersionControl/git_version_control.py]: Import appears to be unused

File: ConfigVersionControl/git_version_control.py [hunk 2]
```diff
@@ -121,169 +119,53 @@ def _unlock(self):
...
+                    self.remote.push()
+                    push_interval = PUSH_BASE_INTERVAL
+                    first_failure = True
+
+                except GitCommandError as e:
+                    # Most likely issue connecting to server, increase timeout, notify if it's the first time
+                    push_interval = PUSH_RETRY_INTERVAL
+                    if first_failure:
+                        print_and_log("Unable to push config changes, will retry in %i seconds"
+                                      % PUSH_RETRY_INTERVAL, "MINOR")
```
Comment (Tom-Willemsen) [inline ConfigVersionControl/git_version_control.py]: Use `.format()` to format strings

File: ConfigVersionControl/git_version_control.py (hunk 1 above)
Comment (ThomasLohnert) [reply ConfigVersionControl/git_version_control.py]: It's not, I'm using `os.path` in a few places here. Not sure why it gets highlighted, I have seen a few people asking the same question online but have not come across a satisfying answer so far

**Comment by Tom-Willemsen** [comment]
- Tested adding `index.lock` in my `.git` directory so that it couldn't commit:q - it retried and committed with an appropriate message once I released the lock. 

Code looks good, functionally works.

**Pull request merged**